frontend/src/app/login/page.tsx

```diff
@@ -4,9 +4,14 @@ import React, {createRef, useState} from "react";
 import {Field} from "~/components/ui/field";
 import {cn} from "~/lib/utils";
 import Form from "~/components/ui/form";
-import {validateRequired} from "~/lib/validations";
 import {useRouter} from "next/navigation";
 import useAuthContext from "~/store/use-auth-context";
+import {object, string, ZodError} from "zod";
```

Comment:
`import * as z from "zod"` and use `z.object` etc. because otherwise you are shadowing typescript's `object` and `string`.

Reply:
Done